### PR TITLE
meterfs: added handling of return values.

### PR DESCRIPTION
**Pull request by Diabliz**

### CODE REVIEW

File: meterfs/meterfs.c [hunk 1]
```diff
@@ -116,51 +137,76 @@ void meterfs_checkfs(meterfs_ctx_t *ctx)
...
+		return 0;
 	}
 
 	/* There should be copy of file table at all times. Fix it if necessary */
 	if (!valid0) {
 		LOG_INFO("meterfs: Filetable header #0 is damaged - repairing.");
 		src = ctx->h1Addr;
 		dst = 0;
-		meterfs_eraseFileTable(0, ctx);
+		if((err = meterfs_eraseFileTable(0, ctx)) < 0) {
```
Comment (agkaminski) [inline meterfs/meterfs.c:154]: Small formatting error here

**Review by agkaminski — APPROVED** [review]
Apart from one style mistake LGTM

File: meterfs/meterfs.c (hunk 1 above)
Comment (Diabliz) [reply meterfs/meterfs.c]: Done

**Review by agkaminski — APPROVED** [review]

**Comment by anglov** [comment]
Please remember to update all flashsrvs using meterfs to conform new meterfs api.
```
flashsrv.c:162:34: warning: assignment to 'int (*)(unsigned int)' from incompatible pointer type 'void (*)(unsigned int)' [-Wincompatible-pointer-types]
  162 |  flashsrv_common.ctx.eraseSector = spiflash_eraseSector;

```

**Pull request merged**